src/toast-container.tsx

```diff
@@ -1,217 +1,310 @@
+    };
+
+    renderUnfolded = () => {
+        this.setState({
+            unfolded: !this.state.unfolded
+        });
+    }
+
+    /**
+     * Updates a toast, To use this create you must pass an id to show method first, then pass it here to update the toast.
+     * @param id: string
+     * @param message: string | JSX.Element
+     * @param toastOptions?: ToastOptions
+     * @returns void
+     */
+    update = (
+        id: string,
+        message: string | JSX.Element,
+        toastOptions?: ToastOptions
+    ) => {
+        let msg: string | JSX.Element;
+        let options: ToastOptions;
+        if(this.state.foldedToast.id === id) {
+            msg = message !== this.state.foldedToast.message && message !== null ? message : this.state.foldedToast.message;
+            options = toastOptions !== this.state.foldedToast && toastOptions !== null ? {...this.state.foldedToast, ...toastOptions} : {...this.state.foldedToast};
+            debugger;
+            this.setState({ foldedToast: {...this.state.foldedToast, ...options, message: msg} });
+        }
+        else {
+            let foundToast = this.state.toastsHistory.find((t) => t.id === id);
+            // @ts-expect-error
+            msg = message !== undefined ? message : foundToast.message;
+            options = toastOptions ? {...foundToast, ...toastOptions} : {...foundToast}
+            this.setState({ toastsHistory: this.state.toastsHistory.map((t) => t.id === id ? {...t, message: msg, ...options} : t) });
+        };
     };
-    return (
-      <KeyboardAvoidingView
-        behavior={Platform.OS === "ios" ? "position" : undefined}
-        style={[styles.container, style]}
-        pointerEvents="box-none"
-      >
-        {toasts
-          .filter((t) => !t.placement || t.placement === "bottom")
-          .map((toast) => (
-            <Toast key={toast.id} {...toast} />
-          ))}
-      </KeyboardAvoidingView>
-    );
-  }
-
-  renderTopToasts() {
-    const { toasts } = this.state;
-    let { offset, offsetTop } = this.props;
-    let style: ViewStyle = {
-      top: offsetTop || offset,
-      width: width,
-      justifyContent: "flex-start",
-      flexDirection: "column-reverse",
+
+    /**
+     * Returns toast with given ID
+     * @param id 
+     * @returns ToastProps |  undefined
+     * 
+     */
+    getNotificationById = (id: string) => {
+        if(this.state.foldedToast.id === id) return this.state.foldedToast;
+        if(this.state.toastsHistory.length > 0) return this.state.toastsHistory.find((toast) => toast.id === id);
+        return;
+    }
+
+    /**
+     * Removes a toast from stack
+     * @param id: string
+     * @returns void
+     */
+    hide = (id: string) => {
+        if(this.state.foldedToast.id === id) return this.setState({ foldedToast: {...this.state.foldedToast, open: false}})
+        return this.setState({ toastsHistory: this.state.toastsHistory.map((t) => t.id === id ? { ...t, open: false } : t) });
```

Comment:
Basically a good compact statement, but a bit overloaded,
maybe do it like i suggested in the example above